src/util/Spotify.js

```diff
@@ -1,142 +1,181 @@
+        return Promise.reject(error);
+      }
+
+      // The code below will try to convert the response to JSON format
+
+      try {
+        JSONResponse = await response.json();
+
+      } catch (err){
+        error = new Error(`Fail to convert response to JSON format, while ${topic}.`);
+        return Promise.reject(error);
+      }
+
+      
+      let JSONTracks = (JSONResponse.tracks && JSONResponse.tracks.items)
+      ? get(JSONResponse, ['tracks', 'items'], [] ) : console.log(`JSONResponse.tracks was not found`);
+
+      let mappedTracks = map(JSONTracks, track => ({
+
+        id: track.id,
+        name: track.name,
+        artist: get(track, ['artists', 0, 'name'], 'No artist name'),
+        album: get(track, ['album', 'name'], 'No album name'),
+        uri: track.uri 
+
+    }))
+      
+      return Promise.resolve(mappedTracks);
+    },
+
+
+    async savePlaylist(playlistName, trackURI){
+      
+      const topic = `Saving Playlist to Spotify`;
+
+      let response;
+      let error;
+      const headers = {Authorization: `Bearer ${userAccessToken}`};
+      let userID;
+      let playlistID;
+
+      if (!playlistName || !trackURI.length){
+        return Promise.reject(new Error(`${topic} failed. Confirm that the playlist has a name and some songs in it`)).then(()=>{}, error => console.log(error));
+      }
+      
+      
+      
+      /*The code below tries to GET the user's ID from the spotify API */
+
+      try {
+        response = await fetch(`${SPOTIFY_SEARCH_ENDPOINT}/me`, {
+          method: 'GET',
+          headers: headers
+        })
+        .then( response => response.json())
+        .then(JSONResponse => userID = JSONResponse.id)
+
+      } catch (e) {
+        error = new Error(`Network error, while ${topic}.`);
+        return Promise.reject(error);
+      };
+
+      
+      /*The code below tries to POST a new playlist in the user’s account and returns a playlist ID */
+      try {
+        response = await fetch(`${SPOTIFY_SEARCH_ENDPOINT}/users/${userID}/playlists`, {
+          headers: headers,
+          method: 'POST',
+          body: JSON.stringify({name: playlistName})
+          })
+          .then(response => response.json())
+          .then(JSONResponse => playlistID = JSONResponse.id)
+          
+      } catch (e) {
+        error = new Error(`Network error, while posting the PlaylistID during ${topic}.`);
+        return Promise.reject(error);
+      };
+
+
+       /*The code below tries to POST the track URIs to the above playlist*/
+      try {
+        response = await fetch(`${SPOTIFY_SEARCH_ENDPOINT}/users/${userID}/playlists/${playlistID}/tracks`, {
+          headers: headers,
+          method: 'POST',
+          body: JSON.stringify({uris: trackURI})
+          })
+      } catch (e) {
+        error = new Error(`Network error, while posting Track URIs to Playlist during ${topic}.`);
+        return Promise.reject(error);
+      };
+
+      }
+};
+
```

Comment:
well done!

As a next step, I suggest you to read about the module-pattern online, just google it with a js prefix.
Its a better way to expose API.